src/rust/inetstack/protocols/udp/tests.rs

```diff
@@ -403,14 +412,15 @@ fn udp_loop2_ping_pong() -> Result<()> {
         now += Duration::from_micros(1);
 
         // Receive data from Bob.
-        alice.receive(bob.rt.pop_frame())?;
-        let mut pop_future = alice.udp_pop(alice_fd);
-        let (remote_addr, received_buf_b) = match Future::poll(Pin::new(&mut pop_future), &mut ctx) {
-            Poll::Ready(Ok((remote_addr, received_buf_b))) => (remote_addr, received_buf_b),
-            _ => anyhow::bail!("pop should have completed"),
-        };
-        crate::ensure_eq!(remote_addr, bob_addr);
-        crate::ensure_eq!(received_buf_b[..], buf_b[..]);
+        alice.receive(bob.rt.pop_frame()).unwrap();
+        let mut coroutine: Pin<Box<Operation>> = alice.udp_pop(alice_fd);
+        let (remote_addr, received_buf_b): (Option<SocketAddrV4>, DemiBuffer) =
+            match Future::poll(coroutine.as_mut(), &mut ctx) {
+                Poll::Ready((_, OperationResult::Pop(addr, buf))) => (addr, buf),
+                _ => unreachable!("Pop failed"),
+            };
+        assert_eq!(remote_addr.unwrap(), bob_addr);
+        assert_eq!(received_buf_b[..], buf_b[..]);
```

Comment:
This piece of code is repeated several times. We should possibly refactor this into a function.